### ISSUE TITLE
No support for lineBreakMode

### ISSUE THREAD

**Issue by tettoffensive** [Raizlabs/BonMot#111]
It doesn't appear that BonMot has support for lineBreakMode so that you can set truncation to be ... among other options. Unless I am missing something. I can't find that this option is exposed.

**Comment by ZevEisenberg** [comment]
@tettoffensive good catch! This is an accidental omission. If you're up for submitting a pull request, the changes from https://github.com/Raizlabs/BonMot/pull/101 and https://github.com/Raizlabs/BonMot/pull/104/files are essentially the same as what would need to change to support `lineBreakMode`.

**Comment by tettoffensive** [comment]
Thanks. I'm up for it, but don't know if I'll get to it today.

**Comment by ZevEisenberg** [comment]
No rush on my end. You're the first person who's requested it. Let me know if you're not going to be able to get to it; otherwise, I look forward to your PR :smile:

**Comment by ZevEisenberg** [comment]
Hey @tettoffensive, is this still something you're interested in? I'm prepping for a 3.0 release that includes some breaking changes. It would be cool to get this in, if you're interested in working on it soon. If not, It doesn't introduce new breaking changes, so it could always go in later as a point release.

**Comment by tettoffensive** [comment]
@ZevEisenberg I'll try to get this in!

**Comment by tettoffensive** [comment]
@ZevEisenberg I made a PR here: https://github.com/Raizlabs/BonMot/pull/128

On another note, do you have a trick in xcode for your indentation with the BonChains? Where each `.` indents on a newline?

**Comment by ZevEisenberg** [comment]
@tettoffensive thanks for the PR - I'll check it out!

As for indentation, Xcode gives me this for Swift:

```swift
attributedString = BONChain()
    .string("Hello, testing world")
    .font(UIFont.preferredFontForTextStyle(UIFontTextStyleBody))
    .textColor(UIColor.redColor())
    .attributedString
```

And this for Objective-C:

```objc
attributedString = BONChain.new
.string(@"Hello, testing world")
.font([UIFont preferredFontForTextStyle:UIFontTextStyleBody])
.textColor([UIColor redColor])
.attributedString;
```

So, no, my README indentations are lies, damned lies, I tell you! Sorry :disappointed:

**Comment by ZevEisenberg** [comment]
@tettoffensive oops, forgot a wrinkle: for my Objective-C code, I use `clang-format` as outlined at the bottom of this project's README. It's what gives me the nice indentation like this:

```objc
NSAttributedString *attributedString =
    BONChain.new
        .string(@"1234")
        .fontNameAndSize(kBONEBGaramond, kBONPointSize)
        .figureSpacing(BONFigureSpacingTabular)
        .attributedString;
```

I have it set to fail commits unless my format is correct, and it auto-generates a patch that I can use to format the code automatically.

**Comment by ZevEisenberg** [comment]
Fixed in #128.
